shims/src/peschke/shims/resource/Closer.scala

```diff
@@ -0,0 +1,36 @@
+package peschke.shims.resource
+
+import cats.syntax.all._
+import cats.ApplicativeThrow
+import peschke.Complete
+
+import java.io.Closeable
+
+/** Close a resource, which can potentially fail.
+  *
+  * Most of the default instances of this will need a [[Convertible]] in the
+  * implicit scope that can lift a [[Throwable]] into `F`
+  */
+trait Closer[F[_], R] {
+  def close(resource: R): F[Complete]
+}
+
+trait CloserJavaInstances {
+  implicit def closerForJavaAutoClosable[F[_], J <: AutoCloseable]
+    (implicit AT: ApplicativeThrow[F])
+    : Closer[F, J] = Closer.wrappingSideEffect(_.close())
+}
+
+trait CloserJavaIOInstances extends CloserJavaInstances {
+  implicit def closerForJavaClosable[F[_], J <: Closeable]
+    (implicit AT: ApplicativeThrow[F])
+    : Closer[F, J] = Closer.wrappingSideEffect(_.close())
+}
+
+object Closer extends CloserJavaIOInstances {
+  def apply[F[_], R](implicit C: Closer[F, R]): C.type = C
+
+  def wrappingSideEffect[F[_], R](close:       R => Unit)
+                                 (implicit AT: ApplicativeThrow[F])
+    : Closer[F, R] = resource => AT.catchNonFatal(close(resource)).as(Complete)
```

Comment:
I think this scalafmt formatting is hard to read. Would be nice to read if formatted like this:

```scala
def wrappingSideEffect[F[_], R](close: R => Unit)(implicit AT: ApplicativeThrow[F]): Closer[F, R] =
  ....
```

or

```scala
def wrappingSideEffect[F[_], R](close: R => Unit)(
                                                        implicit AT: ApplicativeThrow[F]
): Closer[F, R] =
  ....
```

```scala
def wrappingSideEffect[F[_], R](
  close: R => Unit
)(implicit AT: ApplicativeThrow[F]): Closer[F, R] =
  ....
```


Also, consider max column width to 120 if not already configured so.

Reply:
Yeah, I was kind of surprised at how these formatted.

I'm going to revisit this later, I just didn't want to trigger a reformat across the repo as part of this commit.